doc/examples/features_detection/plot_scattering_classification.py

```diff
@@ -0,0 +1,259 @@
+            |                 60000   |  0.51   |
+            +-------------------------+---------+
+
+- **CIFAR10 database**:
+ This is a very challenging database with
+ 10 classes and 32x32 color images. The code presented here can
+ reproduce the results obtained in Table 1 (Trans., order 1)
+ of [2]_, more specifically it can obtain 73.6% success for m=1
+ and 40000 images in the training set.
+
+**Warning**: For integration reasons, the code uses very few images
+in the datasets. To obtain full performance, you need
+to change the variable num_images to 10000 at least.
+
+.. [1] Bruna, J., Mallat, S. 'Invariant Scattering Convolutional Networks'.
+IEEE TPAMI, 2012.
+
+.. [2] Oyallon, E. et Mallat, S. 'Deep Roto-translation Scattering
+ for Object Classification'. CVPR 2015
+
+.. [3] https://en.wikipedia.org/wiki/Support_vector_machine
+"""
+#######################
+# MNIST database
+# Modules, MNIST database and functions for scattering feature generation
+import time as time
+from keras.datasets import mnist
+import numpy as np
+from skimage.filters.filter_bank import multiresolution_filter_bank_morlet2d
+from skimage.feature.scattering import scattering
+from sklearn.svm import SVC
+from sklearn.pipeline import make_pipeline
+from sklearn.metrics import accuracy_score
+from sklearn.preprocessing import MinMaxScaler, Normalizer, StandardScaler
+
+
+def load_images_mnist(px=32, num_images=500000):
+    (X_train_sm, y_train), (X_test_sm, y_test) = mnist.load_data()
+    X_train_sm = X_train_sm[0:num_images, :, :]
+    y_train = y_train[0:num_images]
+    X_test_sm = X_test_sm[0:num_images, :, :]
+    y_test = y_train[0:num_images]
+    num_images_ta = X_train_sm.shape[0]
+    num_images_te = X_test_sm.shape[0]
+    # set size to a power of 2 (32) with a zero pad border
+    X_train = np.zeros((num_images_ta, px, px))
+    X_train[:, 3:31, 3:31] = X_train_sm
+    X_test = np.zeros((num_images_te, px, px))
+    X_test[:, 3:31, 3:31] = X_test_sm
+    X_train = X_train.astype('float32')
+    X_test = X_test.astype('float32')
+    return X_train, y_train, X_test, y_test
+
+
+def load_scattering(X_train, X_test, px=32, J=3, L=8, m=2,
+                    sigma_phi=0.6957, sigma_xi=0.8506):
+    # Create filters
+    Filters, lw = multiresolution_filter_bank_morlet2d(px, J=J, L=L,
+                                                       sigma_phi=sigma_phi,
+                                                       sigma_xi=sigma_xi)
+    # Generate Training set
+    scatterings_train, u, scat_tree = scattering(X_train, Filters, m=m)
+    # Generate Testing set
+    t_scats = time.time()
+    scatterings_test, u, scat_tree = scattering(X_test, Filters, m=m)
+    time_per_image = (time.time() - t_scats) / scatterings_test.shape[0]
+    print('Scattering coefficients computed as ',
+          str(time_per_image), ' secs. per image')
+    return scatterings_train, scatterings_test
+
+
+def gethomogeneus_datast(X, y, d, n):
```

Comment:
explain what this does.
Probably `sklearn.cross_validation.StratifiedShuffleSplit` could have done the job.